src/wasm/extract.cpp

```diff
@@ -329,6 +330,8 @@ std::string extractor::extract(const std::string& list_json) {
   auto files = input["files"];
   std::string lang;
   std::vector<const processed_file*> selected_files;
+  json ret;
```

Comment:
In C++ we just declare variables near their first use. There is no need to declare+define them so early. You could just create it right before `return` (even two times, before each `return`!).